lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -347,18 +347,17 @@ static Value promoteOperand(OpBuilder &builder, Location loc, Value operand,
 
 // promote operands of dot op if the existing combination is not natively
 // supported.
-static void decomposeMixedModeDotOp(ModuleOp mod) {
-  mod.walk([](tt::DotOp dotOp) -> void {
+static void decomposeMixedModeDotOp(ModuleOp mod, int computeCapability) {
+  mod.walk([=](tt::DotOp dotOp) -> void {
     auto D = dotOp.getD();
     OpBuilder builder(dotOp);
     Type AElType = dotOp.getA().getType().getElementType();
     Type promoteType;
     NvidiaMmaEncodingAttr mmaLayout =
         D.getType().getEncoding().dyn_cast<NvidiaMmaEncodingAttr>();
     if (mmaLayout) {
-      bool isNativeHopperFP8 =
-          AElType.isFloat8E5M2() || AElType.isFloat8E4M3FNUZ();
-      if (!isNativeHopperFP8 || (isNativeHopperFP8 && mmaLayout.isHopper()))
+      bool isNativeFP8 = AElType.isFloat8E5M2() || AElType.isFloat8E4M3FNUZ();
+      if (!isNativeFP8 || (isNativeFP8 && computeCapability >= 89))
```

Comment:
thinking more about it, since sm90 doesn't have native mmav2 fp8 I think we should still promote in this case

Reply:
In order to let CI run through, still promote the input when it is not mma v3 for sm90. However, it should be noted that for sm89, fp8 mma will have slight precision difference, which can also be confirmed in the cutlass test.

Reply:
```python
import cutlass
from cutlass.backend.utils.device import device_cc

print_module = True

plan = cutlass.op.Gemm(element=torch.float8_e5m2, element_C=torch.float32, element_D=torch.float32, element_accumulator=torch.float32,
                    layout_A=cutlass.LayoutType.RowMajor, layout_B=cutlass.LayoutType.ColumnMajor,
                    layout_C=cutlass.LayoutType.RowMajor, kernel_cc=89)

# Create input/output tensors in FP8
A = torch.randn((128, 256)).to(torch.float8_e5m2).to("cuda")
B = torch.randn((256, 256)).to(torch.float8_e5m2).to("cuda")
C, D = [torch.zeros((128, 256)).to(torch.float32).to("cuda") for _ in range(2)]

# Run the GEMM
plan.run(A, B, C, D, print_module=print_module)

# Reference FP16
plan = cutlass.op.Gemm(element=torch.float16, element_C=torch.float32, element_D=torch.float32, element_accumulator=torch.float32,
                    layout_A=cutlass.LayoutType.RowMajor, layout_B=cutlass.LayoutType.ColumnMajor,
                    layout_C=cutlass.LayoutType.RowMajor, kernel_cc=89)

A = A.to(torch.float16)
B = B.to(torch.float16)
ref_C, ref_D = [torch.zeros((128, 256)).to(torch.float32).to("cuda") for _ in range(2)]

plan.run(A, B, ref_C, ref_D, print_module=print_module)

print(D)
print(ref_D)
torch.testing.assert_close(ref_D, D)
```
and it tells us,
```shell
Mismatched elements: 31630 / 32768 (96.5%)
Greatest absolute difference: 0.008544921875 at index (79, 15) (up to 1e-05 allowed)
Greatest relative difference: 4.794270992279053 at index (106, 42) (up to 1.3e-06 allowed)
```

Reply:
Do you have any other suggestions? @ThomasRaoux